sdk/keyvault/Azure.Security.KeyVault.Keys/api/Azure.Security.KeyVault.Keys.netstandard2.0.cs

```diff
@@ -532,6 +536,46 @@ public partial class KeyResolver : Azure.Core.Cryptography.IKeyEncryptionKeyReso
         public virtual System.Threading.Tasks.Task<Azure.Security.KeyVault.Keys.Cryptography.CryptographyClient> ResolveAsync(System.Uri keyId, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
     }
     [System.Runtime.InteropServices.StructLayoutAttribute(System.Runtime.InteropServices.LayoutKind.Sequential)]
+    public partial struct KeyVaultContext
```

Comment:
Overall, I'm liking the shape, but I'm still not clear on what this is for. If you have to create a `CryptographyClient` to get an `RSA` or `ECDsa`, what does this context provider. Is it just so you can get back information about the key and/or certificate used so, for example, you can store the information for later? Why wouldn't just a key ID and/or certificate ID be enough?

That said, why not just flatten this onto the KeyVault* classes below?